sulis_core/src/util.rs

```diff
@@ -110,6 +111,16 @@ pub fn gen_rand<T: SampleUniform + Sized>(min: T, max: T) -> T {
     rand::thread_rng().gen_range(min, max)
 }
 
+pub fn gen_rand_weight<'a, T, V>(items: &'a [(T, V)]) -> &'a T
```

Comment:
This is displaying quite different behavior than the old.  It seems to be completely ignoring "none" as a value for an adjective.  For example, loot list level1_weapons always generates an adjective2 whereas it should only do less than 50% of the time for most weapons.

Reply:
You mean you observe this change? strange, the "none" check is in the caller which is in loot_list(), will look into it..

Reply:
If I understand what you are intending correctly, the level 1 weapons and armor loot lists should only have a 5% chance of generating an unidentified item based on the data?  But, in my play testing, they seem to generate one every time.  And the behavior is similar where "none" should be the most common for the other adjectives but you never see none.  Your implementation seems to be skipping the probability of the first entry in the adjectives list.